src/MANYBODY/pair_airebo.cpp

```diff
@@ -3094,72 +3092,54 @@ double PairAIREBO::gSpline(double costh, double Nij, int typei,
 double PairAIREBO::PijSpline(double NijC, double NijH, int typei, int typej,
                              double dN2[2])
 {
-  int x,y,i,done;
-  double Pij,coeffs[16];
-
-  for (i = 0; i < 16; i++) coeffs[i]=0.0;
+  int x,y;
+  double Pij;
 
   x = 0;
   y = 0;
   dN2[0] = 0.0;
   dN2[1] = 0.0;
-  done = 0;
+  Pij = 0.0;
 
-  // if inputs are out of bounds set them back to a point in bounds
+  if (typei == 1) return Pij;
 
-  if (typei == 0 && typej == 0) {
-    if (NijC < pCCdom[0][0]) NijC=pCCdom[0][0];
-    if (NijC > pCCdom[0][1]) NijC=pCCdom[0][1];
-    if (NijH < pCCdom[1][0]) NijH=pCCdom[1][0];
-    if (NijH > pCCdom[1][1]) NijH=pCCdom[1][1];
+  if (typej == 0) {
 
-    if (fabs(NijC-floor(NijC)) < TOL && fabs(NijH-floor(NijH)) < TOL) {
-      Pij = PCCf[(int) NijC][(int) NijH];
-      dN2[0] = PCCdfdx[(int) NijC][(int) NijH];
-      dN2[1] = PCCdfdy[(int) NijC][(int) NijH];
-      done = 1;
-    }
-    if (done == 0) {
-      x = (int) (floor(NijC));
-      y = (int) (floor(NijH));
-      for (i = 0; i<16; i++) coeffs[i] = pCC[x][y][i];
-      Pij = Spbicubic(NijC,NijH,coeffs,dN2);
-    }
-  }
-
-  // if inputs are out of bounds set them back to a point in bounds
+    // if inputs are out of bounds set them back to a point in bounds
 
-   if (typei == 0 && typej == 1){
-     if (NijC < pCHdom[0][0]) NijC=pCHdom[0][0];
-     if (NijC > pCHdom[0][1]) NijC=pCHdom[0][1];
-      if (NijH < pCHdom[1][0]) NijH=pCHdom[1][0];
-      if (NijH > pCHdom[1][1]) NijH=pCHdom[1][1];
+    if (NijC <  pCCdom[0][0]) NijC=pCCdom[0][0];
+    if (NijC >= pCCdom[0][1]) NijC=pCCdom[0][1]-TOL;
+    if (NijH <  pCCdom[1][0]) NijH=pCCdom[1][0];
+    if (NijH >= pCCdom[1][1]) NijH=pCCdom[1][1]-TOL;
+    x = (int) floor(NijC);
```

Comment:
I don't like this solution, because it changes the program behaviour for some points (even if minor).
In particular it 1) changes the point at which the bicubic spline is evaluated and 2) makes the following `if` checking for nodal points useless.
For example, the value of PijSpline at (0, 4, 0, 1) is changed from a pure 0 to 1.4210854715202004e-14, and it is likely that additional values are changed as well along the boundary due to (1).
IMO the cleaner solution (and how I've solved this in my code in the past) is to do `x = (int) floor(NijC)` and then `if (NijC == pCCdom[0][1]) x -= 1;`. This leaves `NijC` and `NijH` unchanged.
I feel like that is a lot more explicit in saying "there are N nodal points and N-1 spline patches".